11-parallelism/src/lib.rs

```diff
@@ -0,0 +1,56 @@
+//! 11 Parallelism
+//! --------------
+//!
+//! Welcome to eleven step of this Rust workshop.
+//!
+//! This step focuses on basic multi threading API in Rust
+//!
+//! ## Thread
+//!
+//! To create a new thread, we call the `thread::spawn` function and pass it a **closure**.
+//!
+//! NOTE:
+//! Calling join on the handle blocks the thread currently running
+//! until the thread represented by the handle terminates.
+//!
+//! ```rust
+//! use std::thread;
+//!
+//! let other_thread = thread::spawn(|| {
+//!     return 1 + 1
+//! });
+//!
+//! let res = other_thread.join().unwrap();
+//! println!("res: {}", res);
+//! ```
+//!
+//! Get more infos: [rust-doc/thread](https://doc.rust-lang.org/std/thread/struct.Thread.html).
+//! 
+//! 
+//! ## Message Passing to Transfer Data Between Threads
+//! 
+//! A approach to ensuring safe concurrency is message passing.
+//! Rust’s standard library provides an implementation of channel.
+//! The channel has two components: a transmitter and a receiver.
+//! 
+//! NOTE: A channel is close if either the transmitter or receiver is dropped.
+//! 
+//! The `send` function takes **ownership** of its parameter, and when the value is moved, 
+//! the receiver takes ownership of it. This stops us from accidentally editing
+//! the value again after sending it; the ownership system checks that everything is okay at compile time.
+//! 
+//! ```rust
+//! use std::thread;
+//! use std::sync::mpsc::channel;
+//! 
+//! let (sender, receiver) = channel();
+//! thread::spawn(move|| {
```

Comment:
May be add some hints about `move` keyword on closure declaration

Reply:
That's a good point, I'll add that in a `NOTE:`

Reply:
Done